xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-api/src/main/java/org/xwiki/security/authentication/api/AuthenticationResourceReference.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.security.authentication.api;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.xwiki.resource.AbstractResourceReference;
+import org.xwiki.resource.ResourceType;
+import org.xwiki.stability.Unstable;
+
+/**
+ * A dedicated {@link org.xwiki.resource.ResourceReference} to perform authentication related actions.
+ * See {@link AuthenticationAction} for the possible actions.
+ *
+ * @version $Id$
+ * @since 13.0RC1
+ */
+@Unstable
+public class AuthenticationResourceReference extends AbstractResourceReference
+{
+    /**
+     * Identifier of the resource type also used for associated components.
+     */
+    public static final String RESOURCE_TYPE_ID = "authenticate";
```

Comment:
"authenticate" does not feel right, but I don't have a perfect fit to propose. I'm wondering if it should not be more about "user" instead.

Reply:
Note that the Maven module is called "security-authentication" and IMO it's really about handling action related to the authentication. Now I chose "authenticate" instead of "authentication" because the other actions are verbs not nouns.